python/ray/util/spark/cluster_init.py

```diff
@@ -1022,8 +1022,9 @@ def _setup_ray_cluster_internal(
     if num_cpus_worker_node is not None and num_cpus_worker_node <= 0:
         raise ValueError("Argument `num_cpus_worker_node` value must be > 0.")
 
+    # note: spark.task.resource.gpu.amount config might be fractional value like 0.5
     num_spark_task_gpus = int(
-        spark.sparkContext.getConf().get("spark.task.resource.gpu.amount", "0")
+        float(spark.sparkContext.getConf().get("spark.task.resource.gpu.amount", "0"))
```

Comment:
We should raise an error if `"spark.task.resource.gpu.amount"` is float given ray node doesn't support fractional gpu?

Reply:
Discussed offline. No need to raise error, instead, we can document the behavior.